### PR TITLE
feat(2710): add Alias column to dashboard collection ,allow pipeline admin to customize name to their pipelines

### PR DESCRIPTION
**Pull request by banusha2**
## Context

To add `alias` column to the dashboard collection UI. In order for the admins to have the ability to customize preferred names for the pipelines. We need to add another aliasName column in the Pipeline DB schema.

## Objective

Pipeline admin can define their preferred names for their pipelines.
Re-name can be shown under the options tab for pipeline level preference
Users can search and add pipeline customized name or git repo name to their pipeline collection. 

## References
refer to: https://github.com/screwdriver-cd/screwdriver/issues/2710

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: models/pipeline.js [hunk 1]
```diff
@@ -33,6 +33,11 @@ const MODEL = {
 
     name: Scm.repoName.optional(),
 
+    aliasName: Joi.string()
+        .max(26)
```
Comment (tkyi) [inline models/pipeline.js:37]: should this be longer?

File: migrations/20220611-add-column-alias-to-pipeline.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const table = `${prefix}pipelines`;
+
+module.exports = {
+    up: async (queryInterface, Sequelize) => {
+        await queryInterface.sequelize.transaction(async transaction => {
+            await queryInterface.addColumn(
+                table,
+                'aliasName',
+                {
+                    type: Sequelize.TEXT('medium')
```
Comment (tkyi) [inline migrations/20220611-add-column-alias-to-pipeline.js:15]: Sequelize.STRING()?

File: models/pipeline.js [hunk 3]
```diff
@@ -135,7 +140,8 @@ module.exports = {
                 'prChain',
                 'parameters',
                 'subscribedScmUrlsWithActions',
-                'settings'
+                'settings',
+                'pipelineAliasName'
```
Comment (tkyi) [inline models/pipeline.js:144]:
```suggestion
                'aliasName'
```

File: models/pipeline.js (hunk 1 above)
Comment (banusha2) [reply models/pipeline.js:37]: No,I guess because it is  alias name

**Review by tkyi — CHANGES_REQUESTED** [review]

File: models/pipeline.js [hunk 4]
```diff
@@ -135,7 +140,8 @@ module.exports = {
                 'prChain',
                 'parameters',
                 'subscribedScmUrlsWithActions',
-                'settings'
+                'settings',
+                'aliasName'
```
[the remark this answers is not in the document]
Comment (banusha2) [reply models/pipeline.js:144]: Yes

[the remark this answers is not in the document]
Comment (tkyi) [reply models/pipeline.js:144]: so you've added `aliasName` to the `get` on this line
do you also need to add `aliasName` to the `update` here: https://github.com/screwdriver-cd/data-schema/blob/master/models/pipeline.js#L159-L161?

[the remark this answers is not in the document]
Comment (banusha2) [reply models/pipeline.js:144]: Yes ,we have to add aliasName to the update.Thanks for pointing out this.

**Comment by tkyi** [comment]
Can you also add the field to mock data for a test?

File: models/pipeline.js [hunk 5]
```diff
@@ -135,7 +135,7 @@ module.exports = {
                 'prChain',
                 'parameters',
                 'subscribedScmUrlsWithActions',
-                'settings'
+                'settings',
```
Comment (tkyi) [inline models/pipeline.js:138]:
```suggestion
                'settings'
```

File: test/data/pipeline.update.yaml [hunk 6]
```diff
@@ -3,6 +3,7 @@ checkoutUrl: git@github.com:screwdriver-cd/data-schema.git#branch
 rootDir: src/app/test
 autoKeysGeneration: false
 settings:
+    aliasName: scwdvr-cd
```
Comment (tkyi) [inline test/data/pipeline.update.yaml:6]:
```suggestion
```

File: test/data/pipeline.get.yaml [hunk 7]
```diff
@@ -30,6 +30,7 @@ settings:
     metricsDowntimeJobs:
             - 123
             - 456
+    aliasName: scwdvr-cd
```
Comment (tkyi) [inline test/data/pipeline.get.yaml:33]:
```suggestion
```

File: migrations/20220611-add-column-alias-to-pipeline.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+/* eslint-disable new-cap */
```
Comment (tkyi) [inline migrations/20220611-add-column-alias-to-pipeline.js:1]: can delete this file if you're no longer adding aliasName to the pipeline model

Comment (banusha2) [reply migrations/20220611-add-column-alias-to-pipeline.js:1]: Yes, I have question, will it work fine if we are directly adding a new field in settings which is present in pipeline model?

File: test/data/pipeline.update.yaml (hunk 6 above)
Comment (banusha2) [reply test/data/pipeline.update.yaml:6]: I didn;t get it why we want remove `aliasName` from here?

Comment (tkyi) [reply test/data/pipeline.update.yaml:6]: oops misread
ignore this comment

File: migrations/20220611-add-column-alias-to-pipeline.js (hunk 8 above)
Comment (tkyi) [reply migrations/20220611-add-column-alias-to-pipeline.js]: yeah should be captured in `settings` field
https://github.com/screwdriver-cd/data-schema/blob/master/migrations/20201007-add-column_settings_to_pipelines.js

**Comment by banusha2** [comment]
@adong @tkyi could it be merged now?

**Review by tkyi — APPROVED** [review]

**Comment by tkyi** [comment]
Need to fix some test errors:
```
11:37:53 /sd/workspace/src/github.com/screwdriver-cd/data-schema/config/settings.js
11:37:53   35:32  error  Insert `⏎············`                                                                                                prettier/prettier
11:37:53   36:1   error  Insert `····`                                                                                                         prettier/prettier
11:37:53   37:1   error  Replace `········.example('scwdvr-cd').optional(),` with `············.example('scwdvr-cd')⏎············.optional()`  prettier/prettier
11:37:53   37:41  error  Unexpected trailing comma                                                                                             comma-dangle
11:37:53 
11:37:53 /sd/workspace/src/github.com/screwdriver-cd/data-schema/test/config/regex.test.js
11:37:53   6:26  warning  Arrow function has too many lines (428). Maximum allowed is 150  max-lines-per-function
11:37:53 
11:37:53 /sd/workspace/src/github.com/screwdriver-cd/data-schema/test/plugins/scm.test.js
11:37:53   7:22  warning  Arrow function has too many lines (235). Maximum allowed is 150  max-lines-per-function
11:37:53 
11:37:53 ✖ 6 problems (4 errors, 2 warnings)
11:37:53   4 errors and 0 warnings potentially fixable with the `--fix` option.
```
https://cd.screwdriver.cd/pipelines/12/builds/805190/steps/test

**Review by tkyi — APPROVED** [review]

**Pull request merged**